test/integration/consul-container/test/resource/grpc_forwarding_test.go

```diff
@@ -10,27 +10,36 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
 
 	libcluster "github.com/hashicorp/consul/test/integration/consul-container/libs/cluster"
 	libtopology "github.com/hashicorp/consul/test/integration/consul-container/libs/topology"
 )
 
 const (
 	RESOURCE_FILE_PATH_ON_HOST      = "../../../../../command/resource/testdata/demo.hcl"
+	CLIENT_CERT_ON_HOST             = "../../../../client_certs/client.crt"
+	CLIENT_KEY_ON_HOST              = "../../../../client_certs/client.key"
+	ROOT_CA_ON_HOST                 = "../../../../client_certs/rootca.crt"
 	RESOURCE_FILE_PATH_ON_CONTAINER = "/consul/data/demo.hcl"
+	CLIENT_CERT_ON_CONTAINER        = "/consul/data/client.crt"
+	CLIENT_KEY_ON_CONTAINER         = "/consul/data/client.key"
+	ROOT_CA_ON_CONTAINER            = "/consul/data/rootca.crt"
 )
 
 func TestClientForwardToServer(t *testing.T) {
 	type operation struct {
-		action       func(*testing.T, libcluster.Agent, string) (int, string)
+		action       func(*testing.T, libcluster.Agent, string, bool) (int, string)
 		includeToken bool
 		expectedCode int
 		expectedMsg  string
 	}
 	type testCase struct {
-		description string
-		operations  []operation
-		aclEnabled  bool
+		description    string
+		operations     []operation
```

Comment:
Unless you're planning to use multiple operations per test, you can collapse this into the `testCase` since there is only ever a single `operation`.

Reply:
makes sense, changed it from slice to single `operation`

Reply:
Why even have a separate `operation`?

Reply:
This is just for readability, do you think it's better for grouping them together into one big embedded struct?